### ISSUE TITLE
General protection fault ip

### ISSUE THREAD

**Issue by francoism90** [gerbera/gerbera#522]
```
traps: gerbera[242601] general protection fault ip:7ff6df6418c2 sp:7ff6c7ffde98 error:0 in libupnp.so.13.1.0[7ff6df62d000+1c000]
```
This happens after updating Gerbera to support the latest support libupnp. Note: I'm using the use ipv6 switch.

**Comment by whyman** [comment]
That is not nice... Can you get a traceback at all? Try running Gerbera under `gdb` and run `bt` when it crashes out, it looks like something is going wrong in libupnp.

Please post that along with an error log at debug level,, thanks!

**Comment by marillat** [comment]
I see the same with upnpn 1.10.1 and grebera 1.3.4
Of course the crash come from upnp
```
Thread 10 "gerbera" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe1ffb700 (LWP 56202)]
UpnpExtraHeaders_get_resp (p=p@entry=0x63696e6f73646572) at ../upnp/inc/TemplateSource.h:334
334	../upnp/inc/TemplateSource.h: Aucun fichier ou dossier de ce type.
(gdb) bt 
#0  0x00007ffff7f4c1d0 in UpnpExtraHeaders_get_resp (p=p@entry=0x63696e6f73646572) at ../upnp/inc/TemplateSource.h:334
#1  0x00007ffff7f42e74 in http_MakeMessage (buf=buf@entry=0x7fffe1ff9bc0, http_major_version=1, http_minor_version=1, fmt=0x7ffff7f556e1 "XcsCc", fmt@entry=0x7ffff7f556d6 "RNTGLDstcSEXcsCc")
    at src/genlib/net/http/httpreadwrite.c:1516
#2  0x00007ffff7f45f17 in process_request (RespInstr=0x7fffe1ff9c50, alias=0x7fffe1ff9c00, filename=0x7fffe1ff9be0, headers=0x7fffe1ff9bc0, rtype=<synthetic pointer>, req=<optimized out>)
    at src/genlib/net/http/webserver.c:1320
#3  0x00007ffff7f45f17 in web_server_callback (parser=0x7fffe1ffa310, req=<optimized out>, info=0x7fffe1ffa280) at src/genlib/net/http/webserver.c:1557
#4  0x00007ffff7f3bcdf in dispatch_request (hparser=0x7fffe1ffa310, info=0x7fffe1ffa280) at src/genlib/miniserver/miniserver.c:157
#5  0x00007ffff7f3bcdf in handle_request (args=0x7fffdc003c90) at src/genlib/miniserver/miniserver.c:230
#6  0x00007ffff7f33175 in WorkerThread (arg=<optimized out>) at src/threadutil/ThreadPool.c:576
#7  0x00007ffff7f75fb7 in start_thread () at /lib/x86_64-linux-gnu/libpthread.so.0
#8  0x00007ffff6feb2cf in clone () at /lib/x86_64-linux-gnu/libc.so.6
```

**Comment by francoism90** [comment]
@whyman It doesn't crash when downgrading libupnp 1.8.5 to 1.8.4 and recompiling Gerbera.
I'll try to provide a crash dump later this week.

**Comment by KarlStraussberger** [comment]
The issue can be fixed in libpupnp by a fix https://github.com/mrjimenez/pupnp/pull/122

**Comment by whyman** [comment]
@KarlStraussberger Thanks for the patch!

I've replicated the issue with both 1.8.5 and 1.10.0 :disappointed: 

But the above patch fixes the issue, so we just need to wait for the new versions

**Comment by marillat** [comment]
Fixed for me in linupnp 1.10.1
Thanks to @KarlStraussberger for the patch

**Comment by whyman** [comment]
Lets close this then, resolved with 1.8.6 and 1.10.1
